gateway/modules/eventing/handle_intents.go

```diff
@@ -68,149 +70,157 @@ func (m *Module) processIntents(t *time.Time) {
+				}
+
+				if err := m.fileStore.DoesExists(ctx, m.project, token, filePayload.Path); err != nil {
+
+					// Mark event as cancelled if it document doesn't exist
+					if err := m.crud.InternalUpdate(ctx, m.config.DBAlias, m.project, utils.TableEventingLogs, m.generateCancelEventRequest(eventID)); err != nil {
+						logrus.Errorf("Eventing: Couldn't cancel intent - %s", err.Error())
+					}
+					return
+				}
+
+				// Mark event as staged if document does exist
+				if err := m.crud.InternalUpdate(ctx, m.config.DBAlias, m.project, utils.TableEventingLogs, m.generateStageEventRequest(eventID)); err != nil {
+					logrus.Errorf("Eventing: Couldn't update intent to staged - %s", err.Error())
+					return
+				}
+
+				// Broadcast the event so the concerned worker can process it immediately
+				eventDoc.Status = utils.EventStatusStaged
+				m.transmitEvents(eventDoc.Token, []*model.EventDocument{eventDoc})
+
+			case utils.EventFileDelete:
+				filePayload := model.FilePayload{}
+				_ = json.Unmarshal([]byte(eventDoc.Payload.(string)), &filePayload)
+
+				token, err := m.auth.GetInternalAccessToken()
+				if err != nil {
+					logrus.Errorf("Eventing: Error generating token in intent staging - %s", err.Error())
+					return
+				}
+
+				if err := m.fileStore.DoesExists(ctx, m.project, token, filePayload.Path); err == nil {
+					// Mark the event as cancelled if the object still exists
+					_ = m.crud.InternalUpdate(ctx, m.config.DBAlias, m.project, utils.TableEventingLogs, m.generateCancelEventRequest(eventID))
+					return
+				}
+
+				// Mark the event as staged if the object doesn't exist
+				if err := m.crud.InternalUpdate(ctx, m.config.DBAlias, m.project, utils.TableEventingLogs, m.generateStageEventRequest(eventID)); err == nil {
+					// Broadcast the event so the concerned worker can process it immediately
+					eventDoc.Status = utils.EventStatusStaged
+					m.transmitEvents(eventDoc.Token, []*model.EventDocument{eventDoc})
+				}
 
-	// Create a context with 5 second timeout
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
-
-	// Get the eventID
-	eventID := eventDoc.ID
-
-	switch eventDoc.Type {
-
-	case utils.EventDBCreate:
-		// Unmarshal the payload
-		createEvent := model.DatabaseEventMessage{}
-		_ = json.Unmarshal([]byte(eventDoc.Payload.(string)), &createEvent)
-
-		// Check if document exists in database
-		readRequest := &model.ReadRequest{Operation: utils.One, Find: createEvent.Find.(map[string]interface{})}
-		if _, err := m.crud.Read(ctx, createEvent.DBType, createEvent.Col, readRequest); err != nil {
-
-			// Mark event as cancelled if it document doesn't exist
-			if err := m.crud.InternalUpdate(ctx, m.config.DBAlias, m.project, utils.TableEventingLogs, m.generateCancelEventRequest(eventID)); err != nil {
-				logrus.Errorf("Eventing: Couldn't cancel intent - %s", err.Error())
 			}
+		case <-m.stopChan:
+			fmt.Println("goroutine processIntent stopped")
 			return
```

Comment:
this will not work as you have a default case, and you are not sending any value to stopchan just directly closing the channel